system/doc/reference_manual/ref_man_records.md

```diff
@@ -55,6 +54,29 @@ definition is placed in an include file.
 > using the syntax described in this section. In earlier releases, it was
 > necessary to use the `m:shell` built-in function `rd/2`.
 
+> #### Change {: .info }
+>
+> In Erlang/OTP 28, variables are allowed in the record definition.
+> If a record definition contains a variable, a function call will be made to
+> initialize the records omitted fields.
+
+```erlang
+-record(r, {a = case my_f() of {X, _} -> X; _ -> ok end,
+            b = case my_g() of {Y, _} -> Y; _ -> ok end}).
+t(X) ->
+   #r{}. %% X will not have any effect on initialization of #r
+```
+
+Variables bound in a field are visible in the fields defined after
+so use with caution.
```

Comment:
I thought that each field would be independent. This seems error-prone.

```suggestion
Variables bound in a field are visible in subsequent fields, so use them with caution.
```

Reply:
Same behavior as if you would define a tuple, chosen to have it this way because of consistency
{r, case my_f() of {X, _} -> X; _ -> ok end, case my_g() of {X, _} -> X; _ -> ok end}